mlptrain/sampling/plumed.py

```diff
@@ -0,0 +1,285 @@
+import mlptrain
+from typing import Sequence
+
+
+class PlumedBias:
+    """Class for storing collective variables and parameters used in biased
+    simulations"""
+
+    def __init__(self,
+                 cvs: Sequence['mlptrain.sampling.plumed.PlumedCV']):
+        """
+        Class for storing collective variables and parameters used in biased
+        simulations, parameters are not initialised with the object and have
+        to be defined seperately using PlumedBias methods.
+
+        -----------------------------------------------------------------------
+        Arguments:
+
+            cvs (Sequence): Sequence of PLUMED collective variable objects
+        """
+
+        self.cvs = cvs
+        self.pace = self.width = self.height = self.biasfactor = None
+
+    def set_metad_params(self,
+                         pace: int,
+                         width: float,
+                         height: float,
+                         biasfactor: float) -> None:
+        """
+        Define parameters used in well-tempered metadynamics.
+
+        -----------------------------------------------------------------------
+        Arguments:
+
+            pace (int): τ_G/dt, interval at which a new gaussian is placed
+
+            width (float): σ, standard deviation (parameter describing the
+                           width) of the placed gaussian
+
+            height (float): ω, initial height of placed gaussians
+
+            biasfactor (float): γ, describes how quickly gaussians shrink,
+                                larger values make gaussians to be placed
+                                less sensitive to the bias potential
+        """
+
+        if not isinstance(pace, int) or pace <= 0:
+            raise ValueError('Pace (τ_G/dt) must be a positive integer')
+        else:
+            self.pace = pace
+
+        if width <= 0:
+            raise ValueError('Gaussian width (σ) must be positive')
+        else:
+            self.width = width
+
+        if height <= 0:
+            raise ValueError('Gaussian height (ω) must be positive')
+        else:
+            self.height = height
+
+        if biasfactor < 1:
+            raise ValueError('Bias factor (γ) must be larger than one')
+        else:
+            self.biasfactor = biasfactor
+
+    @property
+    def cv_sequence(self) -> str:
+        """String containing names of collective variables separated
+        by commas"""
+        cv_names = (cv.name for cv in self.cvs)
+        return ','.join(cv_names)
+
+
+class PlumedCV:
+    """Parent class containing methods for initialising PLUMED collective
+    variables"""
+
+    def __init__(self,
+                 name: str = None,
+                 atom_groups: Sequence = None,
+                 file_name: str = None):
+        """
+        This class contains methods to initialise PLUMED collective variables
```

Comment:
If it shouldn't be called directly, would it be better to make it a private class?